scripts/makefile_help.rb

```diff
@@ -0,0 +1,55 @@
+# Script that parses description strings (## help) out of the Makefile
+# and also uses the `make --print-data-base` output to expand out targets based on variables
+
+expanded_makefile = `make --dry-run --print-data-base --no-builtin-rules`
+
+# Map of lineno => [[lineno, rule]] run by printing the Makefile database
+# rubocop:disable Layout/FirstArgumentIndentation
+# rubocop:disable Lint/MixedRegexpCaptureTypes
+expanded_targets = expanded_makefile.scan(
+/\n\n(?<target>[^:]+):( (?<sources>[^\n]+))?
+((#  [^\n]+)?\n?)*?
+#  commands to execute \(from `Makefile', line (?<lineno>\d+)\):/m,
+).map { |target, *, lineno| [lineno.to_i, target] }.
+  group_by { |lineno, _target| lineno }
+# rubocop:enable Lint/MixedRegexpCaptureTypes
+# rubocop:enable Layout/FirstArgumentIndentation
+
+raw_makefile = File.readlines('Makefile')
+
+# Map of target => [comment, lineno] fom Makefile
+# target might have variables like $(HOST)
+target_comments = raw_makefile.map.with_index.select do |line, _lineno|
+  line =~ / ## /
+end.flat_map do |line, lineno|
+  targets, rest = line.chomp.split(':', 2)
+  _sources, comment = rest.split(' ## ')
+
+  targets.split(' ').map { |target| [target, [comment, lineno + 2]] }
+end.to_h
+
+# Map of target => comment
+cleaned_up_targets = {}
+
+target_comments.each do |target, (comment, lineno)|
+  cleaned_up_name = if target.include?('$(')
+    # "tmp/$(HOST)-$(PORT).crt" into %r|tmp/.+-.+.crt|
+    rule_regexp = Regexp.new(target.gsub(/(\$\([^)]+\))/, '.+'))
```

Comment:
Capture group necessary?

```suggestion
    # "tmp/$(HOST)-$(PORT).crt" into %r|tmp/.+-.+.crt|
    rule_regexp = Regexp.new(target.gsub(/\$\([^)]+\)/, '.+'))
```

Any time I see a `.+`, I also think it might be wise to make it greedy:

```suggestion
    # "tmp/$(HOST)-$(PORT).crt" into %r|tmp/.+?-.+?.crt|
    rule_regexp = Regexp.new(target.gsub(/\$\([^)]+\)/, '.+?'))
```